# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MIR.h
# Commit: 375186706cae
# Full Hash: 375186706caeb4d480bb8c7be78a349601a3e7c1
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2021-02-13 09:52:34
# Regressor Bug: 1688033
# File Overlap Count: 1
# Description:
#   Bug 1688033: Recover CreateArgumentsObject on bailout r=nbp
#   
#   The original version of this patch only recovered CreateArgumentsObject on bailout if it didn't escape, but I removed that code based on discussion in #warpbuilder.
#   
#   This code is tested by `/ion/dce-with-rinstructions.js`. Prior to this patch, it failed with `--scalar-replace-arguments`.
# ==============================================================================

diff -r adaf366a60fa -r 375186706cae js/src/jit/MIR.h
--- a/js/src/jit/MIR.h	Fri Feb 12 20:28:47 2021 +0000
+++ b/js/src/jit/MIR.h	Fri Feb 12 20:28:48 2021 +0000
@@ -3112,6 +3112,10 @@
   AliasSet getAliasSet() const override { return AliasSet::None(); }
 
   bool possiblyCalls() const override { return true; }
+
+  [[nodiscard]] bool writeRecoverData(
+      CompactBufferWriter& writer) const override;
+  bool canRecoverOnBailout() const override { return true; }
 };
 
 class MGetArgumentsObjectArg : public MUnaryInstruction,